src/server/game/Entities/Player/Player.cpp

```diff
@@ -18667,37 +18691,68 @@ bool Player::Satisfy(AccessRequirement const* ar, uint32 target_map, bool report
     return true;
 }
 
-bool Player::CheckInstanceLoginValid(Map* map)
+bool Player::IsInstanceLoginGameMasterException() const
 {
-    if (!map->IsDungeon() || IsGameMaster())
-        return true;
-
-    if (map->IsRaid())
+    if (CanBeGameMaster())
     {
-        // cannot be in raid instance without a group
-        if (!GetGroup())
-            return IsInstanceLoginGameMasterException();
+        ChatHandler(GetSession()).SendSysMessage(LANG_INSTANCE_LOGIN_GAMEMASTER_EXCEPTION);
+        return true;
     }
     else
-    {
-        // cannot be in normal instance without a group and more players than 1 in instance
-        if (!GetGroup() && map->GetPlayersCountExceptGMs() > 1)
-            return IsInstanceLoginGameMasterException();
-    }
-
-    // do checks for satisfy accessreqs, instance full, encounter in progress (raid), perm bind group != perm bind player
-    return sMapMgr->CanPlayerEnter(map->GetId(), this, true) || IsInstanceLoginGameMasterException();
+        return false;
 }
 
-bool Player::IsInstanceLoginGameMasterException() const
+bool Player::CheckInstanceValidity(bool /*isLogin*/)
 {
-    if (CanBeGameMaster())
-    {
-        ChatHandler(GetSession()).SendSysMessage(LANG_INSTANCE_LOGIN_GAMEMASTER_EXCEPTION);
+    // game masters' instances are always valid
+    if (IsGameMaster())
```

Comment:
No, because it would produce a Heisenbug. Game master capability is checked [over here](https://github.com/TrinityCore/TrinityCore/pull/16110/files#diff-5589ee47b3a7ad220cff105a8fe7eac3R17179).